packages/firebase_ml/example/test_driver/firebase_model_manager.dart

```diff
@@ -7,24 +7,57 @@ part of 'firebase_ml.dart';
 void firebaseModelManagerTest() {
   group('$FirebaseModelManager', () {
     final FirebaseModelManager modelManager = FirebaseModelManager.instance;
-    final String MODEL_NAME = 'image_classification';
+    final MODEL_NAME = 'myModelName';
+    final INVALID_MODEL_NAME = 'invalidModelName';
 
-    test('downloadModel', () async {
-      FirebaseCustomRemoteModel model = FirebaseCustomRemoteModel(MODEL_NAME);
-      assert(model != null);
+    test('download model and get its file', () async {
+      final model = FirebaseCustomRemoteModel(MODEL_NAME);
 
-      FirebaseModelDownloadConditions conditions =
-          FirebaseModelDownloadConditions(requireWifi: true);
+      final conditions = FirebaseModelDownloadConditions(
+          androidRequireWifi: true, iosAllowCellularAccess: false);
 
       await modelManager.download(model, conditions);
-      expect(model.modelHash, isNotNull);
 
-      var isModelDownloaded = await modelManager.isModelDownloaded(model);
+      final isModelDownloaded = await modelManager.isModelDownloaded(model);
+
       expect(isModelDownloaded, isTrue);
 
-      var modelFile = await modelManager.getLatestModelFile(model);
+      final modelFile = await modelManager.getLatestModelFile(model);
+
       expect(modelFile, isNotNull);
       expect(modelFile.path.contains(MODEL_NAME), isTrue);
     });
+
+    test('throw an error when model is not downloaded', () async {
+      final model = FirebaseCustomRemoteModel(INVALID_MODEL_NAME);
+
+      final conditions = FirebaseModelDownloadConditions();
+
+      if (Platform.isAndroid) {
+        expect(
+            modelManager.download(model, conditions),
+            throwsA(isA<PlatformException>().having((e) => e.toString(),
+                'message', contains('Failed to schedule the download task'))));
+      }
+
+      if (Platform.isIOS) {
+        expect(
+            modelManager.download(model, conditions),
+            throwsA(isA<PlatformException>().having((e) => e.toString(),
+                'message', contains('The operation couldn’t be completed'))));
```

Comment:
Is this the error message that the iOS SDK provides? I'm wondering if it could be more informative.

Reply:
Yes, it is. Unfortunately, unless we wrap it with our own error message (which I have not seen in other plugins), that's what we get.